include/psa/crypto.h

```diff
@@ -2253,6 +2319,18 @@ typedef struct {
  *                            specifying the public exponent. The
  *                            default public exponent used when \p extra
  *                            is \c NULL is 65537.
+ *                          - For an DSA key (\p type is
+ *                            #PSA_KEY_TYPE_DSA_KEYPAIR), \p extra is an
+ *                            optional structure specifying the key domain
```

Comment:
I find this description unclear: it looks like you can pass `extra`, use `set_key_domain_parameters`, or do neither. In fact you need to do at least one, and the last one applies.

> For a DSA key (`\p type` is `#PSA_KEY_TYPE_DSA_KEYPAIR`) or a finite-field Diffie-Hellman key (`p type` is `#PSA_KEY_TYPE_DH_KEYPAIR`), `extra` is a pointer to a buffer containing domain parameters in DER representation. The format of `buffer` is the same as the input to `psa_set_key_domain_parameters()`. If the application has previously called `psa_set_key_domain_parameters()` on `\p handle`, it may pass an empty buffer (`` `extra_length = 0` ``), in which case the key generation will use the previously set domain parameters.